bc_obps/service/report_service.py

```diff
@@ -107,6 +109,19 @@ def save_report_operation(cls, report_version_id: int, data: ReportOperationIn)
 
         return report_operation
 
+    @classmethod
+    def get_regulated_products_by_version_id(cls, version_id: int) -> List[RegulatedProductOut]:
+        # Fetch the ReportOperation by version ID
+        report_operation = ReportOperation.objects.get(report_version_id=version_id)
+
+        # Retrieve associated regulated products
+        regulated_products = report_operation.regulated_products.all()
+
+        # Map to the output schema (assuming RegulatedProductOut has a 'name' field)
+        return [
+            RegulatedProductOut(id=product.id, name=product.name, unit=product.unit) for product in regulated_products
+        ]
+
```

Comment:
Django ninja will do that for us automatically, no need to do it here. This function should just return a list (or queryset) of RegulatedProduct objects